### ISSUE TITLE
linter: formatter reports utf-8 index while eslint reports in utf-16

### ISSUE THREAD

**Issue by hi-ogawa** [oxc-project/oxc#9110]
### What version of Oxlint are you using?

0.15.10

### What command did you run?

pnpm dlx oxlint -D no-debugger repro.js

### What does your `.oxlintrc.json` config file look like?

None

### What happened?

The code of `repro.js`

```js
"🤨"; debugger;
```

The commands output

```sh
$ pnpm dlx eslint --no-config-lookup --rule 'no-debugger: error' repro.js

/home/hiroshi/code/others/oxc/repro.js
  1:7  error  Unexpected 'debugger' statement  no-debugger

✖ 1 problem (1 error, 0 warnings)


$ pnpm dlx oxlint -D no-debugger repro.js

  × eslint(no-debugger): `debugger` statement is not allowed
   ╭─[repro.js:1:9]
 1 │ "🤨"; debugger;
   ·       ─────────
   ╰────
  help: Delete this code.

Found 0 warnings and 1 error.
Finished in 5ms on 1 file with 97 rules using 16 threads.
```

I'm not sure if this has been discussed, but bringing this up just in case as I'm wondering whether terminal output is fine with column index in utf-8. For example, on Vsocde's terminal, clicking `repro.js:1:9` jumps to the 3rd character `b` of `debugger`.

Interestingly, language server protocol seems to have a knowledge about encoding, so it's reported correctly on Vscode extension.

![Image](https://github.com/user-attachments/assets/7943dffb-c854-41d5-8cf5-755b284f47a5)

**Comment by Boshen** [comment]
This is not trivial to fix. Close as won't fix for now given people don't usually notice this.
